solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -452,23 +453,30 @@ private void setBasicAuthHeader(SolrRequest solrRequest, Request req) {
   private Request makeRequest(SolrRequest solrRequest, String collection)
       throws SolrServerException, IOException {
     Request req = createRequest(solrRequest, collection);
-    req.header(HttpHeader.ACCEPT_ENCODING, null);
```

Comment:
I don't know the implications of moving the lines before/after `setListeners` into the renamed `decorateRequest`. Clearing the `ACCEPT_ENCODING` and adding `REQ_PRINCIPAL_KEY` changes the request?

Reply:
`decorateRequest()` method was added to make sure the consistent for decorating a request between `initOutStream()` and normal `request()`